man/zram-generator.md

```diff
@@ -0,0 +1,57 @@
+zram-generator(8) -- Systemd unit generator for zram swap devices
+=================================================================
+
+## SYNOPSIS
+
+`/lib/systemd/system-generators/zram-generator` `TARGET_DIR` [*2RGET_DIR* *3RGET_DIR*]<br />
+`/lib/systemd/system-generators/zram-generator` --setup-device `DEVICE`<br />
+`/lib/systemd/system-generators/zram-generator` --reset-device `DEVICE`
+
+## DESCRIPTION
+
+`zram-generator` is a generator that creates zram devices and systemd units to configure, format, and enable swap on them.
+
+The generator will be invoked by systemd early at boot, the generator will then:
+
+  1. read configuration files from *{/etc,/lib}/systemd/zram-generator.conf[.d]* (see zram-generator.conf(5) for details);
+  2. if systemd-detect-virt(1) exists and returns non-*0*, exit, since continuing would mean changing the host system;
+  3. generate systemd.swap(5) units into `TARGET_DIR` for all configured devices and symlink them into *swap.target.wants/*;
+  4. if any devices are enabled, and the zram module isn't loaded (*/sys/class/zram-control* doesn't exist), invoke *modprobe(8) zram*;
```

Comment:
I don't think we want to describe this with this level of detail. The implementation is likely to change, and then we'd have to update the man page each time. Maybe just say "generate appropriate units"?

Reply:
Updated

Reply:
I've simplified this a bit, but I've left the `TARGET_DIR` intact, since it's highlit in the SYNOPSIS (and this reinforces that we don't use the 2rget or 3rget), and said "make `swap.target` depend on them", which i think is the happiest marriage (anything more is implementation detail, anything less is unclear re: how (if) these are pulled in and when)